EmberMate/ember_mateApp.swift

```diff
@@ -105,3 +105,24 @@ struct ember_mateApp: App {
         return "\(minutes)min"
     }
 }
+
+private struct OpenSettingsBackportKey: EnvironmentKey {
+    static let defaultValue: () -> Void = {
+        NSApp.sendAction(Selector(("showSettingsWindow:")), to: nil, from: nil)
+    }
+}
+
+extension EnvironmentValues {
+    fileprivate var openSettings_backport: () -> Void {
+        get {
+            if #available(macOS 14.0, *) {
+                return { [openSettings] in
+                    openSettings()
+                }
+            } else {
+                return self[OpenSettingsBackportKey.self]
+            }
+        }
+        set { self[OpenSettingsBackportKey.self] = newValue }
+    }
+}
```

Comment:
Admittedly swift isn't my primary language. What do we gain by doing the check implementation this way, over just updating the `ContextMenu.swift#39` line to:

```swift
if #available(macOS 14.0, *) {
  self.openSettings()
} else {
 NSApp.sendAction(Selector(("showSettingsWindow:")), to: nil, from: nil)
}
```

Reply:
You can't even create the `openSettings` property on macOS 13, since you can't mark stored properties with `@available`. There might be a better way than this to do it though, I'm not sure.

Reply:
Ah, makes sense, assuming this works on macOS 13, its fine with me

I ran through it on macOS 15, but didn't confirm 14 and 13 work. Were you able to test this? or should we reach out to the original reporter?

Reply:
I wasn't able to test it directly on macOS 13 either, I just faked it by hardcoding `if false` instead of the `#available` check.

Reply:
Since its just a backwards compatibility update, shipping it as is seems fine to me. Thanks for throwing this together!